db/compaction/compaction_job.cc

```diff
@@ -850,38 +851,55 @@ Status CompactionJob::VerifyOutputFiles() {
           /*largest_compaction_key=*/nullptr,
           /*allow_unprepared_value=*/false);
       auto s = iter->status();
-
-      if (s.ok() && paranoid_file_checks_) {
-        OutputValidator validator(cfd->internal_comparator(),
-                                  /*_enable_hash=*/true);
-        for (iter->SeekToFirst(); iter->Valid(); iter->Next()) {
-          s = validator.Add(iter->key(), iter->value());
-          if (!s.ok()) {
-            break;
+      if (s.ok()) {
+        // Check for remote/local compaction and verify_output_option flags
+        const bool should_verify =
+            (subcompaction_state.compaction_job_stats.is_remote_compaction &&
+             (verify_output_option &
+              VerifyOutputOptions::kEnableForRemoteCompaction)) ||
+            (!subcompaction_state.compaction_job_stats.is_remote_compaction &&
+             (verify_output_option &
+              VerifyOutputOptions::kEnableForLocalCompaction));
+
+        if (should_verify) {
+          if (verify_output_option &
+              VerifyOutputOptions::kVerifyBlockChecksum) {
+            s = table_reader_ptr->VerifyChecksum(
```

Comment:
If we are also iterating, a lot of this work is redundant (but not filter blocks). Can we avoid it? Perhaps we can have a variant of VerifyChecksum that only verifies metadata blocks? (I'm not too concerned about double-verifying index blocks.)